cotainr/tests/conftest.py

```diff
@@ -102,8 +102,13 @@ def factory_mock_input():
     """
 
     def create_mock_input(fixed_user_input=None):
+        if isinstance(fixed_user_input, list):
+            inputs = iter(fixed_user_input)
+
         def mock_input(prompt):
             print(prompt, end="")
+            if isinstance(fixed_user_input, list):
+                return next(inputs)
             return fixed_user_input
 
         return mock_input
```

Comment:
It's hard to follow what's happening here, especially since the new behavior (that you can pass in a list of inputs and it will be iterated over (until it runs out)) is not documented.

Also, it's a bit weird to me that the `isinstance(fixed_user_input, list)` case is repeated in the inner and outer functions, and that you read from `inputs`, not `fixed_user_input`...

If it helps, `itertools.repeat()` is a handy way to get an infinitely repeating iterable of a value...

```python
def create_mock_input(value: str | List[str] = ""):
    value_iter = itertools.repeat(value) if isinstance(value, str) else iter(value)

    def mock_input(prompt):
        print(prompt, end="")
        return next(value_iter)

    return mock_input
```

Reply:
Maybe we are looking for something along the lines of:
```python
    def create_mock_input(fixed_user_input=None):
        if fixed_user_input is None or isinstance(fixed_user_input, str):
            inputs = iter([fixed_user_input)])
        else:
            inputs = iter(fixed_user_input)

        def mock_input(prompt):
            print(prompt, end="")
            return next(inputs)

        return mock_input
```
with a bit of documentation about what this does

Reply:
I think `None` is a bad default value (which is why my suggestion above replaces it with an empty string) – after all, a real user with a real keyboard can never get `input()` to return `None`.

@Chroxvi That does a different thing. The previous implementation would always return the same string, yours will return it only once, and then raise `StopIteration`.

Reply:
I have given my interpretation of your inputs here. I agree with christians version as the "StopIteration" should not be hit - if it does then there is something wrong in the test any way.